jsonb/src/main/java/io/avaje/jsonb/core/BasicTypeAdapters.java

```diff
@@ -392,7 +392,8 @@ public Object fromJson(JsonReader reader) {
         case STRING:
           return this.stringAdapter.fromJson(reader);
         case NUMBER:
-          return this.doubleAdapter.fromJson(reader);
+          var d = this.doubleAdapter.fromJson(reader);
+          return (d % 1 == 0) ? d.intValue() : d;
```

Comment:
Looks good. Just wondering if this should be longValue() instead of intValue()?  I'm thinking intValue() is probably right but pondering that.

Reply:
oh yeah, double is bigger than an int